crates/re_ui/src/syntax_highlighting.rs

```diff
@@ -0,0 +1,64 @@
+use re_entity_db::InstancePath;
+use re_log_types::{EntityPath, EntityPathPart};
+
+use egui::{text::LayoutJob, Color32, Style, TextFormat};
+
+pub trait SyntaxHighlighting {
```

Comment:
is it actually syntax highlighting or just "make this an egui text"? `ToLayoutJob` might be just as appropriate

Reply:
Discussed live: we'll probably use this trait for more syntax highlighting stuff in the future, and that's what you want to search for in the code. Still, we can revisit the name later if we want to.